smolyay/basis.py

```diff
@@ -172,6 +178,14 @@ def degree(self):
         """int: Degree of polynomial."""
         return self._degree
 
+    @property
+    def integral_over_domain(self):
+        """float: basis function integrated over the domain."""
+        if self.degree == 1:
+            return 0
+        else:
+            return ((-1)**self.degree + 1)/(1-self.degree**2)
```

Comment:
The first case can cover all the odd degrees evaluating to zero, then the second case can be simplified because degree is even:
```suggestion
        if self.degree % 2 == 1:
            return 0
        else:
            return 2/(1-self.degree**2)
```